src/vivarium_census_prl_synth_pop/data/generate_business_names.py

```diff
@@ -0,0 +1,148 @@
+import sys
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from loguru import logger
+
+from vivarium_census_prl_synth_pop.constants import data_values, paths
+
+
+def generate_business_names_data(n_total_names: str):
+    # loads csv of business names and generates pandas series with random business names
+
+    business_names = pd.read_csv(paths.BUSINESS_NAMES_DATA)
+    bigrams = make_bigrams(business_names)
+
+    # Get frequency of business names and find uncommon ones
+    business_names_freq = business_names.location_name.value_counts()
+    real_but_uncommon_names = set(business_names_freq[business_names_freq < 1_000].index)
+
+    # Generate random business names.  Drop duplicates and overlapping names with uncommon names
+    n_total_names = int(n_total_names)  # Make int because of sys args
+    new_names = pd.Series()
+
+    # Generate additional names until desired number of random business names is met
+    while len(new_names) < n_total_names:
+        n_needed = n_total_names - len(new_names)
+        more_names = sample_names(
+            bigrams, n_needed, data_values.BUSINESS_NAMES_MAX_TOKENS_LENGTH
+        )
+        new_names = pd.concat([new_names, more_names]).drop_duplicates()
+        new_names = new_names.loc[~new_names.isin(real_but_uncommon_names)]
+
+    new_names.to_csv(
+        paths.BUSINESS_NAMES_DATA_ARTIFACT_INPUT_PATH,
```

Comment:
I think getting @rmudambi 's input here would be helpful -- depending on how much processing you'll need to do on this, i think it makes sense to put it in share, not the project repo?